### PR TITLE
MTT-3026 feat: Lobby and Authentication services to use async await

### PR DESCRIPTION
**Pull request by pdeschain**
### Description

The API for our lobby sdk wrapper has been converted to async await pattern, and all the call-sites were adapted accordingly.

### Issue Number(s)

MTT-3026

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] JIRA ticket ID is in the PR title or at least one commit message
 - [ ] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

File: Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs [hunk 1]
```diff
@@ -62,87 +64,113 @@ void OnDestroy()
...
+                }
 
-            void OnSuccess(QueryResponse qr)
-            {
+                await m_LobbyServiceFacade.RetrieveAndPublishLobbyListAsync();
                 UnblockUIAfterLoadingIsComplete();
             }
-
-            void OnFailure()
+            catch (Exception)
```
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs:122]: we can use try finally instead here

Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs:122]: or just remove catch all Exception

Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs:122]: and let this crash if there's an unexpected exception

File: Assets/BossRoom/Scripts/Shared/Net/UnityServices/Auth/AuthenticationServiceFacade.cs [hunk 2]
```diff
@@ -18,27 +18,29 @@ void InjectDependencies(IPublisher<UnityServiceErrorMessage> unityServiceErrorMe
...
+                {
+                    await AuthenticationService.Instance.SignInAnonymouslyAsync();
+                }
+            }
+            catch (Exception e)
             {
-                await AuthenticationService.Instance.SignInAnonymouslyAsync();
+                Debug.LogError($"AsyncRequest threw an exception. Call stack before async call:\n{currentTrace}\n");
+                Debug.LogException(e);
+                var reason = $"{e.Message} ({e.InnerException?.Message})";
```
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Shared/Net/UnityServices/Auth/AuthenticationServiceFacade.cs:38]: these shouldn't be needed anymore

**Review by SamuelBellomo — APPROVED** [review]

**Review by fernando-cortez — APPROVED** [review]

**Pull request merged**